File: backend/app/routers/explore.py
from fastapi import APIRouter

from app.schemas.trip import ExploreRequest, ExploreResponse
from app.services import trip_service

router = APIRouter(prefix="/explore", tags=["explore"])


@router.post("", response_model=ExploreResponse)
async def explore(payload: ExploreRequest) -> ExploreResponse:
    suggestions = await trip_service.suggest_places(
        payload.trip_theme,
        payload.activity,
        payload.climate,
        payload.budget,
        payload.duration,
        payload.location,
        payload.trip_type,
        payload.transport,
    )
    return ExploreResponse(places=suggestions)

File: frontend/src/api/explore.ts
import { apiFetch } from './client'
import type { ExploreResponse } from '../types/api'

export interface ExploreRequest {
  trip_theme: string
  activity: string
  climate: string
  budget: string
  duration: string
  location: string
  trip_type: string
  transport: string
}

export function explore(payload: ExploreRequest) {
  return apiFetch<ExploreResponse>('/explore', { method: 'POST', body: JSON.stringify(payload) })
}
